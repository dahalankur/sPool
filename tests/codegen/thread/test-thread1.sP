# dispatch the print function to another thread, and wait for it to finish 
# so that the main thread will not return before the print function is finished
# executing
Thread_join({println("Hello, I am not in main thread!")})

thread t1 = { println(int_to_string(1 + float_to_int(1.99))) }
Thread_join(t1) # wait for it to be done