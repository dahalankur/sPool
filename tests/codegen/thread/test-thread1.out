Hello, I am not in main thread!
2
